velox/vector/DecodedVector.cpp

```diff
@@ -514,4 +516,35 @@ VectorPtr DecodedVector::wrap(
       rows.end(),
       std::move(data));
 }
+
+const uint64_t* DecodedVector::nulls() {
+  if (allNullsInitialized_) {
+    return allNulls_;
+  }
+
+  if (hasExtraNulls_) {
+    allNulls_ = nulls_;
+  } else if (!nulls_ || size_ == 0) {
+    allNulls_ = nullptr;
+  } else {
+    if (isIdentityMapping_) {
+      allNulls_ = nulls_;
+    } else if (isConstantMapping_) {
+      copiedNulls_.resize(0);
+      copiedNulls_.resize(bits::nwords(size_), bits::kNull64);
+      allNulls_ = copiedNulls_.data();
+    } else {
+      // Copy base nulls.
+      copiedNulls_.resize(bits::nwords(size_));
+      auto* rawCopiedNulls = copiedNulls_.data();
+      for (auto i = 0; i < size_; ++i) {
+        bits::setNull(rawCopiedNulls, i, bits::isBitNull(nulls_, indices_[i]));
+      }
+      allNulls_ = copiedNulls_.data();
+    }
+  }
+
+  allNullsInitialized_ = true;
```

Comment:
Maybe allNulls_ can be an optional so that we dont have to use allNullsInitialzed_ can just check if allNulls_.has_value()

Reply:
That's a good idea.